main.go

```diff
@@ -229,12 +235,52 @@ func visit(url *url.URL) {
 				// 30x but no Location to follow, give up.
 				return
 			}
-			log.Fatal("unable to follow redirect: %v", err)
+			log.Fatalf("unable to follow redirect: %v", err)
 		}
 		visit(loc)
 	}
 }
 
+func responseBody(response io.ReadCloser) string {
+
+	var path string
+	var body string
+	var err error
+	if showBodyEnv := os.Getenv(HTTPSTAT_SHOW_BODY); showBodyEnv != "" {
+		showBody, err = strconv.ParseBool(showBodyEnv)
+		if err != nil {
+			log.Fatalf("unable to parse HTTPSTAT_SHOW_BODY env: %v", err)
+		}
+	}
+
+	if !showBody {
+		tmpfile, err := ioutil.TempFile("", "")
+
+		if err != nil {
+			log.Fatalf("unable to create temporary file: %v", err)
+		}
+
+		if _, err := io.Copy(tmpfile, response); err != nil {
+			log.Fatalf("failed to read response body: %v", err)
+		}
+
+		tmpfile.Close()
+
+		path = tmpfile.Name()
+
+	} else {
+		buf, _ := ioutil.ReadAll(response)
+		body = string(buf) // @TODO limit buffer size
+	}
+
+	if path == "" {
+		return fmt.Sprintf("RESPONSE BODY: %v", body)
```

Comment:
If you moved this logic into the previous block, ie rather than setting path to some value above the taking the decision here, just print out the value of path the function will have less branches and be easier to follow.